pkg/jx/cmd/upgrade_platform.go

```diff
@@ -145,20 +145,32 @@ func (o *UpgradePlatformOptions) Run() error {
 
 	wrkDir := ""
 
+	io := &InstallOptions{}
+	io.CommonOptions = o.CommonOptions
+	io.Flags = o.InstallFlags
+	versionsDir, err := io.cloneJXVersionsRepo(o.Flags.VersionsRepository)
+	if err != nil {
+		return err
+	}
+
 	if targetVersion == "" {
-		io := &InstallOptions{}
-		io.CommonOptions = o.CommonOptions
-		io.Flags = o.InstallFlags
-		versionsDir, err := io.cloneJXVersionsRepo(o.Flags.VersionsRepository)
-		if err != nil {
-			return err
-		}
 		targetVersion, err = LoadVersionFromCloudEnvironmentsDir(versionsDir, configStore)
 		if err != nil {
 			return err
 		}
 	}
 
+	isGitOps, devEnv := o.GetDevEnv()
+	if isGitOps {
+		if devEnv == nil {
+			return fmt.Errorf("no Dev environment found")
+		}
+		if devEnv.Spec.Source.URL == "" {
+			return fmt.Errorf("Dev environment does not have source URL")
+		}
+		return o.upgradePlatformViaGitOps(devEnv, targetVersion, versionsDir, configStore)
```

Comment:
This does not check the current version before raising the PR. I think, the update should be skipped if the `targetVersion` == `Current version`.  The logic which verifies this is below.